src/main/scala/org/zalando/spearheads/innkeeper/Routes.scala

```diff
@@ -66,10 +66,29 @@ class Routes @Inject() (implicit val materializer: ActorMaterializer,
               hasOneOfTheScopes(authenticatedUser)(scopes.READ) {
                 metrics.getRoutes.time {
                   LOG.info("get /routes/")
-                  val chunkedStreamSource = jsonService.sourceToJsonSource(routesService.allRoutes)
+                  parameterMap { parameterMap =>
+                    parameterMap.get("name") match {
+                      case Some(name) => {
 
-                  complete {
-                    HttpResponse(entity = HttpEntity.Chunked(MediaTypes.`application/json`, chunkedStreamSource))
+                        Try(RouteName(name)) match {
+                          case Success(routeName) => {
+                            val chunkedStreamSource = jsonService.sourceToJsonSource(routesService.findByName(routeName))
+
+                            complete {
+                              HttpResponse(entity = HttpEntity.Chunked(MediaTypes.`application/json`, chunkedStreamSource))
+                            }
+                          }
+                          case _ => reject(InvalidRouteNameRejection)
+                        }
+                      }
+                      case None => {
+                        val chunkedStreamSource = jsonService.sourceToJsonSource(routesService.allRoutes)
+
+                        complete {
+                          HttpResponse(entity = HttpEntity.Chunked(MediaTypes.`application/json`, chunkedStreamSource))
+                        }
+                      }
+                    }
```

Comment:
this looks like it could be reduced by a lot of boilerplate with the introduction of a function. Both branches have 5 lines of code, that only differ int the actual call to a service.